src/com/owncloud/android/lib/common/OwnCloudClient.java

```diff
@@ -207,7 +209,11 @@ public int executeMethod(HttpMethod method) throws IOException, HttpException {
         		*/
 	            customRedirectionNeeded = mFollowRedirects;
 	        }
-        
+
+            // Update User Agent
+            HttpParams params = method.getParams();
+            params.setParameter(HttpMethodParams.USER_AGENT, mUserAgent);
```

Comment:
This should be enough. For the buggy GET operations detected we may apply the specific code that's needed, but all the others should need nothing else.